app/assets/javascripts/discourse-common/addon/resolver.js

```diff
@@ -85,252 +66,255 @@ export function buildResolver(baseName) {
+            ]
+          ) {
+            corrected = type + ":" + dashed;
+          }
         }
 
-        // Try with dashes instead of slashes
-        dashed = dasherize(split[1].replace(/\./g, "-"));
-        if (
-          requirejs.entries[appBase + dashed] ||
-          requirejs.entries[adminBase + dashed] ||
-          requirejs.entries[wizardBase + dashed]
-        ) {
-          return split[0] + ":" + dashed;
+        let wasCorrected = false;
+        if (corrected && corrected !== normalized) {
+          this._uncorrectedValues[fullName] = normalized;
+          normalized = corrected;
+          wasCorrected = true;
         }
-      }
-      return this._super(fullName);
-    },
-
-    customResolve(parsedName) {
-      // If we end with the name we want, use it. This allows us to define components within plugins.
-      const suffix = parsedName.type + "s/" + parsedName.fullNameWithoutType,
-        dashed = dasherize(suffix),
-        moduleName = lookupModuleBySuffix(dashed);
-
-      let module;
-      if (moduleName) {
-        module = requirejs(moduleName, null, null, true /* force sync */);
-        if (module && module["default"]) {
-          module = module["default"];
+
+        // Check if we have any other values that normalized to the same thing. In the future,
+        // when we remove this code, these will no longer work as expected.
+        let match = Object.entries(this._normalizeCache).find(
+          ([key, value]) => {
+            return (
+              value === normalized &&
+              (wasCorrected || this._uncorrectedValues[key])
+            );
+          }
+        );
+        if (match) {
+          deprecated(
+            function () {
+              let message = `Both ${fullName} and ${match[0]} normalized to the same value: ${normalized}. In the future they will not normalize to the same thing.`;
+              if (this._uncorrectedValues[fullName]) {
+                message += ` ${fullName} will normalize to ${this._uncorrectedValues[fullName]}`;
+              }
+              if (this._uncorrectedValues[match[0]]) {
+                message += ` ${match[0]} will normalize to ${
+                  this._uncorrectedValues[match[0]]
+                }`;
+              }
+              return message;
+            }.call(this)
+          );
         }
       }
-      return module;
-    },
-
-    resolveWidget(parsedName) {
-      return this.customResolve(parsedName) || this._super(parsedName);
-    },
 
-    resolveAdapter(parsedName) {
-      return this.customResolve(parsedName) || this._super(parsedName);
-    },
+      return normalized;
+    }
 
-    resolveModel(parsedName) {
-      return this.customResolve(parsedName) || this._super(parsedName);
-    },
-
-    resolveView(parsedName) {
-      return this.customResolve(parsedName) || this._super(parsedName);
-    },
+    chooseModuleName(moduleName, parsedName) {
```

Comment:
This is an ember-resolver method that we're hooking into. It always seems best to use the lowest level possible.